src/test/java/sqlline/SqlLineArgsTest.java

```diff
@@ -2225,6 +2228,91 @@ public void testStartupArgsWithoutUrl() {
     }
   }
 
+  /**
+   * Test to Set/Reset confirm.
+   */
+  @Test
+  public void testConfirmFlag() {
+    DispatchCallback callback = new DispatchCallback();
+    SqlLine line = new SqlLine();
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+    line.runCommands(
+            Collections.singletonList("!set confirm true"), callback);
+    assertEquals(Boolean.TRUE, line.getOpts().getConfirm());
+    line.runCommands(
+            Collections.singletonList("!set confirm false"), callback);
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+  }
+
+  /**
+   * Test effects of confirm flag.
+   */
+  @Test
+  public void testConfirmFlagEffects() {
+    new MockUp<sqlline.Commands>() {
+      @Mock
+      private int getUserAnswer(
+              String question, int... allowedAnswers) throws IOException {
+        return 'y';
+      }
+    };
+    SqlLine line = new SqlLine();
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+    assertEquals("default", line.getOpts().getConfirmPattern());
+    String script = "CREATE TABLE dummy_test(pk int);\n"
+            + "INSERT INTO dummy_test(pk) VALUES(1);\n"
+            + "INSERT INTO dummy_test(pk) VALUES(2);\n"
+            + "INSERT INTO dummy_test(pk) VALUES(3);\n"
+            + "DELETE FROM dummy_test;\n"
+            + "DROP TABLE dummy_test;\n";
+    checkScriptFile(script, true,
+            equalTo(SqlLine.Status.OK),
+            containsString(" "));
+    script = "!set confirm true\n" + script;
+    checkScriptFile(script, true,
+            equalTo(SqlLine.Status.OK),
+            containsString(" "));
+
+  }
+
+  /**
+   * Test effects of confirm flag.
+   */
+  @Test
+  public void testConfirmPattern() {
+    DispatchCallback callback = new DispatchCallback();
+    SqlLine line = new SqlLine();
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+    line.runCommands(
+            Collections.singletonList("!set confirmPattern "
+                    + "\"^(?i:(TRUNCATE|ALTER))\""), callback);
+    assertEquals("^(?i:(TRUNCATE|ALTER))", line.getOpts()
+            .getConfirmPattern());
+    line.runCommands(
+            Collections.singletonList("!set confirmPattern default"), callback);
+    assertEquals("default", line.getOpts().getConfirmPattern());
+
+  }
+
+  /**
+   * Test for illegal regex in confirmPattern
+   */
+  @Test
+  public void testExceptionConfirmPattern() {
+    DispatchCallback callback = new DispatchCallback();
+    SqlLine line = new SqlLine();
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
```

Comment:
Could you please use custom output stream here, e.g. like at `sqlline.SqlLineArgsTest#testSetWrongColorScheme`
it will keep output clearer in case of success tests